framework/src/modules/chain/logic/block/index.js

```diff
@@ -0,0 +1,210 @@
+const BlockOld = require('./block');
+const BlockBFT = require('./block_bft');
+
+class Block {
+	constructor(ed, schema, transaction, bftUpgradeHeight) {
```

Comment:
This class is behaving like proxy to the right block logic. You can explore using javascript proxy class that may be helpful in this context. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy